src/hooks/useInput.tsx

```diff
@@ -0,0 +1,17 @@
+import { useState } from 'react';
+
+const useInput = (initialText: string) => {
+  const [inputText, setInputText] = useState(initialText);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    setInputText(e.target.value);
+  };
```

Comment:
```suggestion
  const handleInputChange = useCallback((e: React.ChangeEvent<HTMLInputElement>): void => {
    setInputText(e.target.value);
  }, []);
```
인풋 처리 관련 코드의 함수는 useCallback을 사용하여 최적화 하는것이 좋습니다.